src/Features/LanguageServer/CommonLanguageServerProtocol.Framework/RequestDispatcher.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics.Contracts;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CommonLanguageServerProtocol.Framework;
+
+/// <summary>
+/// Aggregates handlers for the specified languages and dispatches LSP requests
+/// to the appropriate handler for the request.
+/// </summary>
+public class RequestDispatcher<RequestContextType> : IRequestDispatcher<RequestContextType>
+{
+    protected ILspServices _lspServices;
+
+    public RequestDispatcher(ILspServices lspServices)
+    {
+        throw new NotImplementedException();
+    }
+
+    protected virtual ImmutableDictionary<RequestHandlerMetadata, Lazy<IRequestHandler>> GetRequestHandlers()
```

Comment:
this feels a bit weird.  why not `Lazy<RequestHandlerMetadata, IRequestHandler>`?

Reply:
I don't know if I follow, how can we have a dictionary where the key hasn't been calculated yet?